app/src/main/java/org/schabi/newpipe/fragments/list/search/filter/BaseSearchFilterUiGenerator.java

```diff
@@ -0,0 +1,93 @@
+// Created by evermind-zz 2022, licensed GNU GPL version 3 or later
+
+package org.schabi.newpipe.fragments.list.search.filter;
+
+
+import android.content.Context;
+import android.util.DisplayMetrics;
+import android.util.TypedValue;
+
+import org.schabi.newpipe.R;
+
+import androidx.annotation.NonNull;
+
+import static android.util.TypedValue.COMPLEX_UNIT_DIP;
+import static org.schabi.newpipe.fragments.list.search.filter.SearchFilterLogic.ICreateUiForFiltersWorker;
+import static org.schabi.newpipe.fragments.list.search.filter.SearchFilterLogic.IUiItemWrapper;
+
+/**
+ * The base class to implement the search filter UI for content
+ * and sort filter dialogs eg. {@link SearchFilterDialogGenerator}
+ * or {@link SearchFilterOptionMenuAlikeDialogGenerator}.
+ */
+public abstract class BaseSearchFilterUiGenerator {
+    protected final ICreateUiForFiltersWorker contentFilterWorker;
+    protected final ICreateUiForFiltersWorker sortFilterWorker;
+    protected final Context context;
+    protected final SearchFilterLogic logic;
+
+    protected BaseSearchFilterUiGenerator(
+            @NonNull final SearchFilterLogic logic,
+            @NonNull final Context context) {
+        this.context = context;
+        this.logic = logic;
+        this.contentFilterWorker = createContentFilterWorker();
+        this.sortFilterWorker = createSortFilterWorker();
+    }
+
+    // common helpers
+    public static float dipToPixels(@NonNull final Context ctx, final float dipValue) {
+        final DisplayMetrics metrics = ctx.getResources().getDisplayMetrics();
+        return TypedValue.applyDimension(COMPLEX_UNIT_DIP, dipValue, metrics);
+    }
```

Comment:
You can use `DeviceUtils.dpToPx(int, Context)`

Reply:
done